src/CanvasRenderingContext2d.cc

```diff
@@ -3354,3 +3358,29 @@ Context2d::Ellipse(const Napi::CallbackInfo& info) {
   }
   cairo_set_matrix(ctx, &save_matrix);
 }
+
+#if CAIRO_VERSION >= CAIRO_VERSION_ENCODE(1, 16, 0)
+
+/*
+ * Open and close a link tag
+ */
+
+void
+Context2d::BeginTag(const Napi::CallbackInfo& info) {
+  Napi::String strValue;
+  if (!info[0].ToString().UnwrapTo(&strValue)) return;
+
+  std::string str = strValue.Utf8Value();
+  std::string uri = "uri='" + str + "'";
```

Comment:
Wonder if we need some sort of escaping or validity testing to make sure `str` doesn't contain `'` or any other impermissible characters?

Cairo's docs say URI needs to be an ASCII string; what happens if it's not?

Reply:
yep, escaping `'` and `\` is needed.

I'd probably guess things are going to crash with a non-ascii string, its actually where this is likely very helpful, https://github.com/Automattic/node-canvas/pull/2410